fv3gfs-physics/fv3gfs/physics/stencils/microphysics.py

```diff
@@ -2234,8 +2226,31 @@ def setupm(self):
         self._csmlt = csmlt
         self._cgmlt = cgmlt
         self._ces0 = constants.EPS * es0
+        self._set_timestep(-1.0)
+
+    def _update_timestep_if_needed(self, timestep: float):
+        if timestep != self._timestep:
+            self._set_timestep(timestep=timestep)
+
+    def _set_timestep(self, timestep: float):
+        # Define cloud microphysics sub time step
+        self._mpdt = min(timestep, self.namelist.mp_time)
+        self._rdt = 1.0 / timestep
+        self._ntimes = int(round(timestep / self._mpdt))
+        # Small time step
+        self._dts = timestep / self._ntimes
+        self._dt_rain = self._dts * 0.5
+        self._rdts = 1.0 / self._dts
+        self._dt_evap = 0.5 * self._dts if self.namelist.fast_sat_adj else self._dts
+        self._fac_i2s = 1.0 - np.exp(-self._dts / self.namelist.tau_i2s)
+        self._fac_g2v = 1.0 - np.exp(-self._dts / self.namelist.tau_g2v)
+        self._fac_v2g = 1.0 - np.exp(-self._dts / self.namelist.tau_v2g)
+        self._fac_imlt = 1.0 - np.exp(-0.5 * self._dts / self.namelist.tau_imlt)
+        self._fac_l2v = 1.0 - np.exp(-self._dt_evap / self.namelist.tau_l2v)
+        self._timestep = timestep
 
-    def __call__(self, state: MicrophysicsState):
+    def __call__(self, state: MicrophysicsState, timestep: float):
+        self._set_timestep(timestep)
```

Comment:
This is not supposed to be here right? or is some logic missing here? I see that you have another function called `_update_timestep_if_needed` but I don't see it used.

Reply:
Thanks, fixed it to call `update_timestep_if_needed`.